### ISSUE TITLE
Issues with ipywidgets developer install and npm

### ISSUE THREAD

**Issue by captainsafia** [ipython/ipywidgets#279]
So I'm trying to install ipywidgets in dev mode and I'm experiencing a bit of trouble. I started off by running `pip install -e .` and recieved the following image.

```
rebuilding js and css failed
missing files: ['/Users/safia/dev/ipywidgets/ipywidgets/static/widgets/css/widgets.min.css']
error: [Errno 2] No such file or directory: '/Users/safia/dev/ipywidgets/package.json'
```

Hm. So that's strange. It failed to build and it appears it is looking for `package.json` in the wrong place. So I go into `setup.py` and update line 199 to the following.

```
 return mtime(self.node_modules) < mtime(pjoin(repo_root, 'ipywidgets', 'package.json'))
```

By the way, why is it not this in the first place?

Then I run `pip install -e .` again and get the following error.

```
subprocess.CalledProcessError: Command '['npm', 'run', 'build']' returned non-zero exit status 1
```

Alright, so it looks like `npm run build` didn't do its job well. So I do `cd ipywidgets && npm run build`` and get the following error.

```
> node build_css.js

> lessc --include-path=./static/components ./static/widgets/less/widgets.less ./static/widgets/css/widgets.css
/Users/safia/dev/ipywidgets/ipywidgets/build_css.js:19
    process.stdout.write(p.stdout.toString());
                                 ^

TypeError: Cannot read property 'toString' of null
    at run (/Users/safia/dev/ipywidgets/ipywidgets/build_css.js:19:34)
    at Object.<anonymous> (/Users/safia/dev/ipywidgets/ipywidgets/build_css.js:28:1)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:475:10)
    at startup (node.js:117:18)
    at node.js:951:3
```

Alright, so it looks like there is something up with `build_css.js`, so I dive into the file. I add two `console.logs` to print out the value of `p`, run `node build_css.js`, and start investigating the output.

```
cleancss --source-map --skip-restructuring  -o ./static/widgets/css/widgets.min.css ./static/widgets/css/widgets.css
{ error: 
   { [Error: spawnSync cleancss ENOENT]
     code: 'ENOENT',
     errno: 'ENOENT',
     syscall: 'spawnSync cleancss',
     path: 'cleancss',
     spawnargs: 
      [ '--source-map',
        '--skip-restructuring ',
        '-o',
        './static/widgets/css/widgets.min.css',
        './static/widgets/css/widgets.css' ] },
  status: null,
  signal: null,
  output: null,
  pid: 4846,
  stdout: null,
  stderr: null,
```

So it looks like the troublesome command is the `cleancss` command. Hm. That's weird. `npm install` should have taken care of it. I run `cleancss` on the commandline and get the usual command not found error. So I run `npm install clean-css` and run `cleancss` and it doesn't work. Then I run `npm install -g clean-css` and rerun `node build_css.js` and it works well. Woot!

Then I go back and run `pip install -e .` and the install works well! Woot.

But I'm curious and have destructive tendencies so I removed the `ipywidgets` that I added to the `package.json` path earlier and I rerun the install and it fails. So I'm curious as to why this happens for me but persumably not for anyone else.

This isn't really an issue since I ended up solving the problem in the process of writing up the issue and going through the above steps, but I figured I should leave it here in case other users experience similar issues.

So long story short:
* Is there a way to ensure that `npm install cleancss` installs the command line program too?
* Why is the path for `package.json` in `setup.py` seemingly incorrect?

cc: @jdfreder

**Comment by jdfreder** [comment]
Thanks for digging into this @captainsafia , this is a lot of useful information.  I'll address this in my WIP https://github.com/ipython/ipywidgets/pull/278 .  We recently moved the package.json file into ipywidgets/ipywidgets, it may be that we missed the path in setup.py.

`npm install cleancss` installs the CLI to `node_modules`.  This way it's available to the node scripts, but doesn't clutter the user's global namespace.  If you need to use it, you can by, within the ipywidgets/ipywidgets directory, running `$(npm bin)/cleancss`

cc @SylvainCorlay

**Comment by captainsafia** [comment]
> We recently moved the package.json file into ipywidgets/ipywidgets, it may be that we missed the path in setup.py.

SETUP DOT PIIIIIIEEEE!

>  If you need to use it, you can without a global install by, within the ipywidgets/ipywidgets directory, running $(npm bin)/cleancss

That makes sense. I was trying to run it inside `ipywidgets` instead of `ipywidgets/ipywidgets`. Tricky, very tricky.
